src/Plugin/GraphQL/DataProducer/EntityDefinition/Fields/Status.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\EntityDefinition\Fields;
+
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Field\Entity\BaseFieldOverride;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\graphql\Plugin\GraphQL\DataProducer\DataProducerPluginBase;
+
+/**
+ * Retrieves the "status" property from a given field definition.
+ *
+ * @DataProducer(
+ *   id = "entity_definition_field_status",
+ *   name = @Translation("Entity definition field status"),
+ *   description = @Translation("Return entity definition field status."),
+ *   consumes = {
+ *     "entity_definition_field" = @ContextDefinition("any",
+ *       label = @Translation("Entity definition field")
+ *     )
+ *   },
+ *   produces = @ContextDefinition("string",
+ *     id = @Translation("Entity definition field status")
+ *   )
+ * )
+ */
+class Status extends DataProducerPluginBase {
+
+  /**
+   * Resolves the "status" property.
+   *
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $entity_definition_field
+   *   The entity field definition.
+   *
+   * @return bool
+   *   If the field config is enabled or not.
+   */
+  public function resolve(FieldDefinitionInterface $entity_definition_field): bool {
+    if ($entity_definition_field instanceof BaseFieldDefinition) {
+      /** @var \Drupal\Core\Field\BaseFieldDefinition $entity_definition_field */
+      return TRUE;
+    }
+    elseif ($entity_definition_field instanceof FieldConfig) {
+      /** @var \Drupal\field\Entity\FieldConfig $entity_definition_field */
+      return $entity_definition_field->status();
+    }
+    elseif ($entity_definition_field instanceof BaseFieldOverride) {
+      /** @var \Drupal\Core\Field\Entity\BaseFieldOverride $entity_definition_field */
+      return $entity_definition_field->status();
+    }
+
+    return NULL;
```

Comment:
this will throw a fatal error because the function must return a bool

Reply:
good point!

Reply:
wow, somehow forgot about this pr, fixed now. Please re-check